index.js

```diff
@@ -83,22 +83,26 @@ function getPublicPath(compiler) {
   return compiler.options.output && compiler.options.output.publicPath || '';
 }
 
-function stripTrailingSlash(string) {
-  return string.replace(/\/$/, '');
-}
-
 function createLoaderRules(languages, features, workers, publicPath) {
   if (!languages.length && !features.length) { return []; }
   const languagePaths = languages.map(({ entry }) => entry).filter(Boolean);
   const featurePaths = features.map(({ entry }) => entry).filter(Boolean);
-  const workerPaths = workers.reduce((acc, { label, output }) => Object.assign(acc, {
-    [label]: `${publicPath ? `${stripTrailingSlash(publicPath)}/` : ''}${output}`,
-  }), {});
+  const workerPaths = fromPairs(workers.map(({ label, output }) => [label, output]));
+
   const globals = {
-    'MonacoEnvironment': `(function (paths) { return { getWorkerUrl: function (moduleId, label) { return paths[label]; } }; } )(${
-      JSON.stringify(workerPaths, null, 2)
-    })`,
+    'MonacoEnvironment': `(function (paths) {
+      function stripTrailingSlash(string) {
```

Comment:
Not sure if it's better to have this inside the string or outside and `toString()` it so other parts of the code can re-use this (although nothing else uses it atm).

Reply:
Good point – the `stripTrailingSlash()` function was only there for the `getWorkerUrl()` – now that it's embedded within the string, the outer one should be removed.

I personally think it's better to keep the `stripTrailingSlash()` definition inside the string, seeing as it's clear what's going on (`Function.prototype.toString` always seems like a bit of a hack to me…)

Reply:
fair point 👍. I think i'm good with this now. 
I was thinking to rename parameter from `string` to `str` to prevent any confusion with reserved keywords potentially.

Reply:
`string` isn't a reserved keyword but I see your point, `str` might make more sense :)